src/api/telemetry/TelemetryCollection.js

```diff
@@ -326,12 +330,14 @@ export class TelemetryCollection extends EventEmitter {
         let domains = this.metadata.valuesForHints(['domain']);
         let domain = domains.find((d) => d.key === timeSystem.key);
 
-        if (domain === undefined) {
-            this._error(ERRORS.TIMESYSTEM_KEY);
+        if (domain !== undefined) {
+            // timeKey is used to create a dummy datum used for sorting
+            this.timeKey = domain.source;
+        } else {
```

Comment:
I think we want to: `this.timeKey = undefined` here in case there is a time system change, we don't hold onto a previous time key value.

Reply:
good catch.  added.